tests/test_director.py

```diff
@@ -252,6 +252,26 @@ def test_05_register_ecu_serial(self):
         False]                 # is_primary
 
 
+    # Expect no registered ECUs or manifests yet.
+    self.assertFalse(inventory.ecu_public_keys)
+    self.assertFalse(inventory.vehicle_manifests[vin])
+    self.assertFalse(inventory.ecu_manifests)
+    self.assertIsNone(inventory.primary_ecus_by_vin[vin])
```

Comment:
How `primary_ecus_bin_vin[vin]` is verified to be empty stood out to me.  Is there a reason why None is used instead of a Boolean, unlike the others?  I didn't dig in deeply, but if all these either store dicts and lists, shouldn't you consistently check for booleanness?

Reply:
Each vehicle has a single entry in primary_ecus_by_vin; rather than a list, it's a string. If that vehicle has no registered Primary, the value is None.